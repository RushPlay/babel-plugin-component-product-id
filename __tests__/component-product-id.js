const { transform } = require('babel-core')
const componentProductId = require('../plugins/component-product-id')

describe('plugins/component-product-id', () => {
  it('transforms createElement component name to LogicalExpression', () => {
    // arrange
    const input = `
      const Component = {}
      React.createElement(Component)
    `
    const options = { plugins: [componentProductId] }

    // act
    const { code } = transform(input, options)

    // assert
    expect(code).toMatchSnapshot()
  })

  it('uses given variable name instead process.env.PRODUCT_ID', () => {
    // arrange
    const input = `
      const Component = {}
      React.createElement(Component)
    `
    const pluginOptions = { productIdSource: '__PROD_ID__' }
    const options = { plugins: [[componentProductId, pluginOptions]] }

    // act
    const { code } = transform(input, options)

    // assert
    expect(code).toMatchSnapshot()
  })

  it('skips other CallExpressions', () => {
    // arrange
    const input = `
      const Component = {}
      React.createClass(Component)
      console.log(Component)
    `
    const options = { plugins: [componentProductId] }

    // act
    const { code } = transform(input, options)

    // assert
    expect(code).toMatchSnapshot()
  })

  it('skips string components', () => {
    // arrange
    const input = `
      React.createClass('div')
    `
    const options = { plugins: [componentProductId] }

    // act
    const { code } = transform(input, options)

    // assert
    expect(code).toMatchSnapshot()
  })
})
