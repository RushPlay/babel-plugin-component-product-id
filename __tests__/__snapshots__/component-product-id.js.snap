// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugins/component-product-id skips other CallExpressions 1`] = `
"
const Component = {};
React.createClass(Component);
console.log(Component);"
`;

exports[`plugins/component-product-id skips string components 1`] = `
"
React.createClass('div');"
`;

exports[`plugins/component-product-id transforms createElement component name to LogicalExpression 1`] = `
"
const Component = {};
React.createElement(Component[process.env.PRODUCT_ID] || Component);"
`;

exports[`plugins/component-product-id uses given variable name instead process.env.PRODUCT_ID 1`] = `
"
const Component = {};
React.createElement(Component[__PROD_ID__] || Component);"
`;
